### PR TITLE
#1377 - Support incremental sync in the mixpanel source

### LINKED ISSUE #1377: Support incremental sync in the mixpanel source
## Tell us about the problem you're trying to solve
Before beginning implementation for this ticket, please first see the Gist describing how to [support incremental sync for singer sources](https://gist.github.com/sherifnada/6970d73d93db52451b0b39fe59002566) in Airbyte.

### PR DESCRIPTION
**Pull request by eugene-kulak**
## What
enable incremental sync in existing Mixpanel singer source

## How
remove `force_full_refresh=True` and update testing samples

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

File: airbyte-integrations/connectors/source-mixpanel-singer/sample_files/sample_config.json [hunk 1]
```diff
@@ -4,5 +4,5 @@
   "attribution_window": "5",
   "project_timezone": "US/Pacific",
   "select_properties_by_default": "true",
-  "start_date": "2019-01-01T00:00:00Z"
+  "start_date": "2020-10-01T00:00:00Z"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mixpanel-singer/sample_files/sample_config.json:7]: Why this change? If there was a corresponding change in the secrets/config.json file, make sure you update it in lastpass.

**Review by sherifnada — APPROVED** [review]
LGTM -- please bump version and I'll merge tomorrow morning.

**Review by sherifnada — CHANGES_REQUESTED** [review]
Woops, didn't realize tests aren't passing. Please request a review again once those are ready.

**Comment by sherifnada** [comment]
FYI the error message from the incremental test suite is: 
```
JUnit Jupiter:PythonTestSource:testIncrementalSyncWithState()
    MethodSource [className = 'io.airbyte.integrations.standardtest.source.PythonTestSource', methodName = 'testIncrementalSyncWithState', methodParameterTypes = '']
    => org.opentest4j.AssertionFailedError: Expected the first incremental sync to produce records ==> expected: <false> but was: <true>
```

The failing test case is `testIncrementalSyncWithState` -- see [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector/standard-source-tests) for more details on this method

File: airbyte-integrations/connectors/source-mixpanel-singer/sample_files/sample_config.json (hunk 1 above)
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-mixpanel-singer/sample_files/sample_config.json:7]: to get rid of the warning about too old start_date (1 year is max), my initial thought was that this is the cause of broken test, but it is not
